### ISSUE TITLE
Enhancement: feedback and session comments

### ISSUE THREAD

**Issue by dzhuang** [inducer/relate#581]
Currently, page feedback messages and session comments are hard coded in the database, that might bring result which are not satisfying, when l10n is enabled:

1. Some auto-feedback will not be translated

2. Saved code feedback displayed in the language of student browser.

3. Mixed language in session comments(#575).

This need to deal with literals which need to be formatted, for example:

```python
   "Your answer is somewhat correct. (%.1f%%)" % (correctness * 100)
```

To work around this, feedback message can be store in the form of a list of list.  For example,
```python
    [["Your answer is somewhat correct. (%.1f%%)"，(correctness * 100)]]

    class AnswerFeedback(object):
          ...
          def render_feedback_text(self):
                if not isinstance(self.feedback, list):
               # backward compatibility
                    return _(self.feedback)

               def render_single_feedback(fb):
                    if len(fb) == 1:
                        return _(fb[0])
                   else:
                        return _(fb[0] % fb[1:])
 
                fbs = [render_single_feedback(fb) for fb in self.feedback]
                return "\n".join(fbs)
                    
```

 For session comments,  the field need to be changed to a JsonField, and do the same thing as above.

How about this idea? @inducer

**Comment by inducer** [comment]
I agree that doing something systematic about l10n for session comments seems like a good idea. We would have to decide exactly how much structure we want to impose here. We could create a new "session log" table, with a number of distinct event types (that would be part of an enum, e.g. "closed", "reopened", "regraded"), plus maybe some JSON for unstructured additional information.

I feel that feedback is overall less structured and harder to translate. #353 is something that work on that should consider, but the simplest correct thing here would be to ensure that feedback (to the extent it comes out of code) is always emitted in the site default language.

**Comment by dzhuang** [comment]
I agree.
